spec/fixtures/api/gdk-backend-check/before-test.sh

```diff
@@ -0,0 +1,17 @@
+#!/bin/sh
+
+set -o errexit
+
+# Install the application Groot
+cp ./groot /tmp/
+# Register Groot as desktop application
+desktop-file-install --dir=$HOME/.local/share/applications ./groot.desktop
+
+# Update database of desktop entries
+update-desktop-database ~/.local/share/applications
+
+# Associate 'application/x-groot' mime type with '*.groot' files
+cp ./groot.xml ~/.local/share/mime/packages/
+
+# update the MIME database
+update-mime-database ~/.local/share/mime
```

Comment:
can u verify if these non-trivial commands are blocking? I just wanted to make sure this doesn't cause any race condition when the rest of the test expects these operations to be completed but they are actually in progress

Reply:
From the documentation, it looks like a synchronous operation as the command can print the summary

Reply:
https://linux.die.net/man/1/update-mime-database#:~:text=update%2Dmime%2Ddatabase%20is%20a,from%20the%20X%20Desktop%20Group.

Reply:
Sounds good! Is that applicable for these too:

* update-desktop-database
* desktop-file-install

?

Reply:
@RaisinTen Can you help me understand if these are blocking?
How can we confirm the same?

Reply:
Addressed the rest of the review comments and the PR is ready if we can resolve this comment